core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -70,6 +70,8 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   var proxyUser: String = null
   var principal: String = null
   var keytab: String = null
+  var usePython: Boolean = false
```

Comment:
`isPython` and `usePython` has different semantic. `isPython` means it is pyspark application and the primary resource should be python script or `pyspark-shell`, and we  use `isPython` to figure out the mainClass. While `usePython` doesn't mean it is pyspark application ( it could be scala application but use pyspark internally ), and it is not related with mainClass.